shell/renderSessions/DrawInstancedSession.cpp

```diff
@@ -0,0 +1,268 @@
+             return tex;
+           }
+        )";
+}
+
+static const char* getVulkanVertexShaderSource() {
+  return R"(#version 460
+layout (location=0) in vec2 offset;
+layout (location=0) out vec3 color;
+const vec2 pos[6] = vec2[6](
+    vec2(-0.05f,  0.05f),
+    vec2( 0.05f, -0.05f),
+    vec2( -0.05f, -0.05f),
+    vec2(-0.05f,  0.05f),
+    vec2(0.05f, -0.05f),
+    vec2(0.05f,  0.05f)
+);
+const vec3 col[6] = vec3[6](
+	vec3(1.0, 0.0, 0.0),
+	vec3(0.0, 1.0, 0.0),
+	vec3(0.0, 0.0, 1.0),
+    vec3(1.0, 0.0, 0.0),
+    vec3(0.0, 1.0, 0.0),
+    vec3(0.0, 0.0, 1.0)
+);
+void main() {
+	gl_Position = vec4(pos[gl_VertexIndex] + offset , 0.0, 1.0);
+	color = col[gl_VertexIndex];
+}
+)";
+}
+static const char* getVulkanFragmentShaderSource() {
+  return R"(#version 460
+precision mediump float;
+precision highp int;
+layout (location=0) in vec3 color;
+layout (location=0) out vec4 out_FragColor;
+void main() {
+	out_FragColor = vec4(color, 1.0);
+}
+)";
+}
+
+static std::unique_ptr<IShaderStages> getShaderStagesForBackend(igl::IDevice& device) {
+  switch (device.getBackendType()) {
+  case igl::BackendType::Invalid:
+    IGL_ASSERT_NOT_REACHED();
+    return nullptr;
+  case igl::BackendType::Vulkan:
+    return igl::ShaderStagesCreator::fromModuleStringInput(device,
+                                                           getVulkanVertexShaderSource(),
+                                                           "main",
+                                                           "",
+                                                           getVulkanFragmentShaderSource(),
+                                                           "main",
+                                                           "",
+                                                           nullptr);
+    return nullptr;
+  // @fb-only
+  // @fb-only
+  // @fb-only
+  case igl::BackendType::Metal:
+    return igl::ShaderStagesCreator::fromLibraryStringInput(
+        device, getMetalShaderSource().c_str(), "vertexShader", "fragmentShader", "", nullptr);
+  case igl::BackendType::OpenGL:
+#if IGL_BACKEND_OPENGL
+    auto glVersion =
+        static_cast<igl::opengl::Device&>(device).getContext().deviceFeatures().getGLVersion();
+    if (glVersion > igl::opengl::GLVersion::v2_1) {
+      auto usesOpenGLES =
+          static_cast<igl::opengl::Device&>(device).getContext().deviceFeatures().usesOpenGLES();
+      auto codeVS1 = std::regex_replace(
+          getVulkanVertexShaderSource(), std::regex("gl_VertexIndex"), "gl_VertexID");
+      auto codeVS2 =
+          std::regex_replace(codeVS1.c_str(), std::regex("460"), !usesOpenGLES ? "410" : "300 es");
+
+      auto codeFS = std::regex_replace(
+          getVulkanFragmentShaderSource(), std::regex("460"), !usesOpenGLES ? "410" : "300 es");
+
+      if (usesOpenGLES) {
+        codeVS2 =
+            std::regex_replace(codeVS2.c_str(), std::regex("layout \\(location=0\\) out"), "out");
```

Comment:
No need to use regexps for this. Use this instead (change to `const char*` where appropriate):
```
[[maybe_unused]] void stringReplaceAll(std::string& s,
                                       const std::string& searchString,
                                       const std::string& replaceString) {
  size_t pos = 0;
  while ((pos = s.find(searchString, pos)) != std::string::npos) {
    s.replace(pos, searchString.length(), replaceString);
  }
}
```
Please remove all `<regex>` usage.